ui/store/action-queue/index.test.js

```diff
@@ -0,0 +1,188 @@
+import sinon from 'sinon';
+
+import {
+  callBackgroundMethod,
+  submitRequestToBackground,
+  _setBackgroundConnection,
+} from '.';
+
+// This file tests only MV3 queue scenario
+// MV2 tests are already covered by '../actions.test.js'
+
+jest.mock('../../../shared/modules/mv3.utils', () => {
+  return {
+    isManifestV3: () => true,
+  };
+});
+
+describe('ActionQueue', () => {
+  describe('submitRequestToBackground', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('calls promisified background method if the stream is connected', () => {
+      const background = {
+        connectionStream: {
+          readable: true,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction');
+      expect(background.backgroundFunction.called).toStrictEqual(true);
+    });
+
+    it('does not calls promisified background method if the stream is not connected', () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction');
+      expect(background.backgroundFunction.called).toStrictEqual(false);
+    });
+
+    it('calls promisified background method on stream reconnection', () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction');
+      expect(background.backgroundFunction.called).toStrictEqual(false);
+
+      background.connectionStream = {
+        readable: true,
+      };
+      _setBackgroundConnection(background);
+      submitRequestToBackground('backgroundFunction');
+      expect(background.backgroundFunction.called).toStrictEqual(true);
+    });
+
+    // it('resolves if backgroundFunction called resolves', async () => {
+    //   const background = {
+    //     connectionStream: {
+    //       readable: true,
+    //     },
+    //     backgroundFunction: async () => {
```

Comment:
this was supposed to be a callback style function - it's being wrapped in pify afterall. 
Hours spent here: 2 :D